### PR TITLE
[Impeller] avoid creating multiple concurrent message loops for Andorid Vulkan

### PR DESCRIPTION
**Pull request by dnfield**
Fixes https://github.com/flutter/flutter/issues/127160

We should avoid creating multiple message loops that each have as many threads as processors on the machine.

### CODE REVIEW

**Review by gaaclarke — COMMENTED** [review]
Ah, this is the issue jonah was referring to in the other PR, eh?

This looks mostly good except I wouldn't muddy `DeviceHolder` and follow the pattern we had previously of the clients holding onto a shared pointer to the concurrent task runner.

File: impeller/renderer/backend/vulkan/device_holder.h [hunk 1]
```diff
@@ -12,6 +13,8 @@ class DeviceHolder {
  public:
   virtual ~DeviceHolder() = default;
   virtual const vk::Device& GetDevice() const = 0;
+  virtual const std::shared_ptr<fml::ConcurrentTaskRunner>
+  GetConcurrentWorkerTaskRunner() const = 0;
```
[the remark this answers is not in the document]
Comment (dnfield) [reply impeller/renderer/backend/vulkan/device_holder.h:17]: Fair enough, I'll remove this and still give a shared pointer ot the runner to the pipeline.

I wanted this to help with testing but I can still test it by exposing a getter on the context anyway.

File: impeller/renderer/backend/vulkan/device_holder.h [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 #pragma once
 
+#include "flutter/fml/concurrent_message_loop.h"
```
Comment (gaaclarke) [inline impeller/renderer/backend/vulkan/device_holder.h:7]: You can remove this now.

File: impeller/renderer/backend/vulkan/context_vk.cc [hunk 3]
```diff
@@ -427,6 +434,11 @@ const vk::Device& ContextVK::GetDevice() const {
   return device_.get();
 }
 
+const std::shared_ptr<fml::ConcurrentTaskRunner>
+ContextVK::GetConcurrentWorkerTaskRunner() const {
+  return worker_task_runner_;
+}
+
```
Comment (gaaclarke) [inline impeller/renderer/backend/vulkan/context_vk.cc:441]: We don't need this anymore right?

Comment (dnfield) [reply impeller/renderer/backend/vulkan/context_vk.cc:441]: It's used in the test.

File: impeller/renderer/backend/vulkan/device_holder.h (hunk 2 above)
Comment (dnfield) [reply impeller/renderer/backend/vulkan/device_holder.h]: done

**Review by gaaclarke — APPROVED** [review]
lgtm!

**Pull request merged**